security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -497,7 +500,13 @@ func (sc *SecretManagerClient) generateFileSecret(resourceName string) (bool, *s
 		// Check if the resource name refers to a file mounted certificate.
 		// Currently used in destination rules and server certs (via metadata).
 		// Based on the resource name, we need to read the secret from a file encoded in the resource name.
-		cfg, ok := model.SdsCertificateConfigFromResourceName(resourceName)
+		cfg := security.SdsCertificateConfig{}
+		ok := false
+		if resourceName == security.FileRootSystemCACert {
```

Comment:
It seems to be all these special casing for the `caRootPath` do not follow the existing SDS case switching.

Instead of adding a new parameter to the function, (and 90% callers will now have to pass an empty string), can we model the caroot path as a special `resourceName` instead?


add @shankgan who is more familiar in this code.

Reply:
Made the resourceName check into it's own case and removed the parameter added to the function.